mlflow/store/artifact/artifact_repo.py

```diff
@@ -57,6 +57,10 @@ def list_artifacts(self, path):
         """
         pass
 
+    def _is_directory(self, artifact_path):
```

Comment:
Thanks for adding the generic artifact repo tests in test_artifact_repo.py, I'm wondering how you knew to override the `_is_directory` implementation in the DBFS & FTP artifact repos - should we add a test for `download_artifacts` against an empty directory across the artifact repos? Unfortunately I suspect that'll be a lot of work though (because it requires mocking artifact download & listing logic / we don't have a generic artifact repo test suite that the others extend)

Reply:
Yeah good question, I have added the override for filesystem like repos. I don't think we need to fix this for other repos. E.g. it works fine with s3 implementation. I think we should have a test for all the repos but I that is the responsibility of each repo implementation and I think does not need to be addressed in this PR.